### PR TITLE
Fix lifecycle routing for Claude, Codex, and OpenCode

### PR DESCRIPTION
**Pull request by openwong2kim**
## Summary
- isolate Claude, Codex, and OpenCode lifecycle routing and bridge state with `WMUX_DATA_SUFFIX`
- parse the official Codex turn-complete payload without forwarding prompt or assistant content
- install and refresh the OpenCode plugin, keep `session.idle` non-blocking, and safely fall back from unsupported daemons
- extend `setup-hooks` install/status handling while preserving foreign notify and plugin configuration

## Validation
- `npx tsc --noEmit`
- `npx tsc --noEmit -p tsconfig.daemon.json`
- targeted ESLint: 0 errors / 0 warnings
- targeted Vitest suite: 152 tests passed
- `npm run package`
- production + suffixed daemon routing smoke
- Codex official payload privacy smoke
- OpenCode non-blocking and daemon-to-main fallback smoke
- installed integration status: all up to date

## Summary by CodeRabbit

* **New Features**
  * Added OpenCode lifecycle integration alongside Claude and Codex.
  * Added isolated routing for multiple concurrent integration instances.
  * Added support for official Codex lifecycle events and thread/turn metadata.
  * Added automatic installation, status reporting, and safe refreshing of managed integration assets.
  * Added coordinated setup and removal status reporting across integrations.

* **Bug Fixes**
  * Prevented unrelated or malformed lifecycle events from being processed.
  * Preserved foreign or malformed configuration instead of overwriting it.
  * Improved fallback handling, retries, and durable delivery of lifecycle signals.
  * Reduced sensitive payload details in diagnostic logs.

### CODE REVIEW

**Comment by openwong2kim** [comment]
Addressed the review feedback in `df695be2`:

- treat `notify = []` as an unoccupied Codex notify slot
- preserve legacy root fields in `setup-hooks --json` while adding nested integration status
- resolve the OpenCode plugin directory through absolute `XDG_CONFIG_HOME`, falling back to `~/.config`
- centralize lifecycle asset specs for CLI and app registration
- use one permission ID resolver for asked/replied events

Revalidated both TypeScript configs, targeted ESLint, 152 tests, CLI bundle, and a smoke covering all four behavioral edge cases.

File: src/shared/mcpRegistration.ts [hunk 1]
```diff
@@ -282,11 +297,15 @@ export function registerCodexNotify(home: string, notifyScript: string): Registe
...
+  const slot = inspectNotifySlot(parsed);
+  // A root notify slot that is non-array, contains non-string values, or names
+  // another program is user-owned/unknown. Treat all of those as a conflict so
+  // setup never rewrites a value it cannot prove belongs to wmux.
+  if (slot.present && (!slot.validStringArray || !isWmuxOwnedNotify(slot.notify))) {
+    return { configPath, skipped: 'foreign', wrote: false };
   }
-  if (isWmuxOwnedNotify(notify) && norm(notify![1]) === norm(notifyScript)) {
+  const notify = slot.notify;
+  if (isWmuxOwnedNotify(notify) && norm(notify?.[1] ?? '') === norm(notifyScript)) {
```
[the remark this answers is not in the document]
Comment (openwong2kim) [reply src/shared/mcpRegistration.ts:312]: Fixed in df695be2: empty string arrays now resolve as an unoccupied slot, so registration claims notify = [] and status reports none until installed. Invalid/non-string arrays and real foreign commands remain preserved as conflicts.

**Comment by openwong2kim** [comment]
## [wmux-hermes] Triage Summary

This PR hardens lifecycle routing across all three first-party agent bridges (Claude, Codex, OpenCode) and consolidates their install/refresh machinery into a shared `lifecycleIntegrations` module. The headline change is making every bridge path (`WMUX_DATA_SUFFIX`) instance-aware: tokens, daemon pipe hints, named pipes, log/spool/activity-stamp directories now resolve into `~/.wmux${suffix}` instead of always hitting the production `~/.wmux`, which closes a real dev/prod cross-talk hole where a suffixed deck's signals could land in the production instance. On the Codex side it adopts the official `agent-turn-complete` payload (`type`/`thread-id`/`turn-id`), explicitly drops unrelated lifecycle event types, and carefully forwards only metadata — prompt and assistant content is never logged or copied into the envelope. The OpenCode plugin gains daemon-first dispatch (with the main pipe as a safe fallback), detaches `session.idle` handling from the OpenCode event loop so a wedged endpoint cannot stall the TUI, durable resume-binding spool on failure, and consistent `permission.asked`/`updated`/`replied` correlation across SDK variants.

The supporting work is equally careful: `setup-hooks` now installs and statuses Claude, Codex, and OpenCode integrations independently (a corrupt Claude settings file no longer blocks the other two), the new shared asset installer uses ownership markers plus atomic temp→rename writes and refuses to touch a destination it cannot prove is wmux-owned, and `mcpRegistration.ts` replaces a loose `getNotify` with a stricter slot inspector that treats malformed/non-array/foreign notify values as conflicts rather than silently clobbering them. Validation looks thorough (`tsc --noEmit` on both projects, targeted ESLint clean, 152 targeted Vitest tests, `npm run package`, plus production/suffixed routing, Codex privacy, and OpenCode fallback smoke tests). This is integration-correctness work that closes real cross-instance routing and payload-privacy gaps, so I am tagging it **P2**.
